lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers_test.go

```diff
@@ -1044,6 +1044,108 @@ func TestGetSubscriber(t *testing.T) {
+	assert.NoError(t, err)
+
+	e := echo.New()
+	getSubscriberURL := "/magma/v1/lte/:network_id/subscribers/:subscriber_id"
+	createSubscribersURL := "/magma/v1/lte/:network_id/subscribers_v2"
+	handlers := handlers.GetHandlers()
+	getSubscriber := tests.GetHandlerByPathAndMethod(t, handlers, getSubscriberURL, obsidian.GET).HandlerFunc
+	createSubscribers := tests.GetHandlerByPathAndMethod(t, handlers, createSubscribersURL, obsidian.POST).HandlerFunc
+
+	// Pre: create two APNs
+	_, err = configurator.CreateEntities(
+		"n1",
+		[]configurator.NetworkEntity{
+			{Type: lte.APNEntityType, Key: "apn0"},
+			{Type: lte.APNEntityType, Key: "apn1"},
+		},
+		serdes.Entity,
+	)
+	assert.NoError(t, err)
+
+	// Create two subscribers, one a prefix of the other
+	sub1 := newMutableSubscriber("IMSI1234567890")
+	sub2 := newMutableSubscriber("IMSI123456789000")
+	sub2.Name = "John Doe"
+	payload := subscriberModels.MutableSubscribers{sub1, sub2}
+
+	tc := tests.Test{
+		Method:         "POST",
+		URL:            createSubscribersURL,
+		Payload:        tests.JSONMarshaler(payload),
+		Handler:        createSubscribers,
+		ParamNames:     []string{"network_id"},
+		ParamValues:    []string{"n1"},
+		ExpectedStatus: 201,
+	}
+	tests.RunUnitTest(t, e, tc)
+
+	tc = tests.Test{
+		Method:         "GET",
+		URL:            getSubscriberURL,
+		Handler:        getSubscriber,
+		ParamNames:     []string{"network_id", "subscriber_id"},
+		ParamValues:    []string{"n1", "IMSI1234567890"},
+		ExpectedStatus: 200,
+		ExpectedResult: sub1.ToSubscriber(),
+	}
+	tests.RunUnitTest(t, e, tc)
+
+	// Now create some AGW-reported state for the subscribers
+	// First we need to register a gateway which can report state
+	_, err = configurator.CreateEntity(
+		"n1",
+		configurator.NetworkEntity{Type: orc8r.MagmadGatewayType, Key: "g1", Config: &models.MagmadGatewayConfigs{}, PhysicalID: "hw1"},
+		serdes.Entity,
+	)
+	assert.NoError(t, err)
+	frozenClock := int64(1000000)
+	clock.SetAndFreezeClock(t, time.Unix(frozenClock, 0))
+	defer clock.UnfreezeClock(t)
+
+	icmpStatus := &subscriberModels.IcmpStatus{LatencyMs: f32Ptr(12.34)}
+	ctx := test_utils.GetContextWithCertificate(t, "hw1")
+	test_utils.ReportState(t, ctx, lte.ICMPStateType, "IMSI1234567890", icmpStatus, serdes.State)
+
+	// Sub1 and sub2 differ in their mme states
+	mmeState1 := state.ArbitraryJSON{"mme": "foo"}
+	test_utils.ReportState(t, ctx, lte.MMEStateType, "IMSI1234567890", &mmeState1, serdes.State)
+	mmeState2 := state.ArbitraryJSON{"mme": "foo"}
+	test_utils.ReportState(t, ctx, lte.MMEStateType, "IMSI123456789000", &mmeState2, serdes.State)
+
+	sub1_with_state := sub1.ToSubscriber()
+	sub1_with_state.Monitoring = &subscriberModels.SubscriberStatus{
+		Icmp: &subscriberModels.IcmpStatus{
+			LastReportedTime: frozenClock * 1000,
+			LatencyMs:        f32Ptr(12.34),
+		},
+	}
+	sub1_with_state.State = &subscriberModels.SubscriberState{
+		Mme: mmeState1,
+	}
```

Comment:
- Don't use icmpStatus state at all, get rid of that and just report the mme state
- Call it `sub1` -- also, generally, use camelCase rather than snake_case

Reply:
Sounds good! I renamed it to `sub1ExpectedResult` instead so as there's no confusion about `sub1` being first a `MutableSubscriber` and then `Subscriber`, if that sounds ok to you? @hcgatewood